packages/cactus-cmd-api-server/src/test/typescript/integration/jwt-unprotected-endpoint-authz.test.ts

```diff
@@ -26,102 +26,94 @@ const log = LoggerProvider.getOrCreate({
-    };
-
-    // look Ma, no access token
-    const res1 = await axios.request({
-      data: req1,
-      url: `${apiHost}${ep.getPath()}`,
-      method: ep.getVerbLowerCase() as Method,
-    });
-    t.ok(res1, "stub unprotected action response truthy OK");
-    t.equal(
-      res1.status,
-      StatusCodes.OK,
-      "stub unprotected action response status === 200 OK",
-    );
-    t.equal(typeof res1.data, "object", "typeof res1.data is 'object' OK");
-    t.ok(typeof res1.data.data, "res1.data.data truthy OK");
-    t.ok(
-      typeof res1.data.data.reqBodyrequestId,
-      "res1.data.data.reqBody truthy OK",
-    );
-    t.ok(
-      typeof res1.data.data.reqBody.requestId,
-      "res1.data.data.reqBody.requestId truthy OK",
-    );
-    t.equal(
-      res1.data.data.reqBody.requestId,
-      req1.requestId,
-      "res1.data.requestId === req1.requestId OK",
-    );
-  } catch (ex) {
-    log.error(ex);
-    t.fail("Exception thrown during test execution, see above for details!");
-    throw ex;
-  }
+describe(testCase, () => {
+  let apiServer: ApiServer;
+
+  afterAll(async () => {
+    await apiServer.shutdown();
+  });
+
+  test(testCase, async () => {
+    try {
+      const jwtKeyPair = await generateKeyPair("RS256", {
+        modulusLength: 4096,
+      });
+      const jwtPublicKey = await exportSPKI(jwtKeyPair.publicKey);
+      const expressJwtOptions: expressJwt.Options = {
+        algorithms: ["RS256"],
+        secret: jwtPublicKey,
+        audience: uuidv4(),
+        issuer: uuidv4(),
+      };
+      expect(expressJwtOptions).toBeTruthy();
+
+      const ep = new UnprotectedActionEndpoint({
+        connector: {} as PluginLedgerConnectorStub,
+        logLevel,
+      });
+
+      const authorizationConfig: IAuthorizationConfig = {
+        unprotectedEndpointExemptions: [ep.getPath()],
+        expressJwtOptions,
+        socketIoJwtOptions: { secret: jwtPublicKey },
+      };
+
+      const pluginRegistry = new PluginRegistry();
+      const plugin = new PluginLedgerConnectorStub({
+        logLevel,
+        pluginRegistry,
+        instanceId: uuidv4(),
+      });
+      pluginRegistry.add(plugin);
+
+      const configService = new ConfigService();
+      const apiSrvOpts = await configService.newExampleConfig();
+      apiSrvOpts.authorizationProtocol = AuthorizationProtocol.JSON_WEB_TOKEN;
+      apiSrvOpts.authorizationConfigJson = authorizationConfig;
+      apiSrvOpts.configFile = "";
+      apiSrvOpts.apiCorsDomainCsv = "*";
+      apiSrvOpts.apiPort = 0;
+      apiSrvOpts.cockpitPort = 0;
+      apiSrvOpts.grpcPort = 0;
+      apiSrvOpts.apiTlsEnabled = false;
+      apiSrvOpts.plugins = [];
+      const config = await configService.newExampleConfigConvict(apiSrvOpts);
+
+      apiServer = new ApiServer({
+        config: config.getProperties(),
+        pluginRegistry,
+      });
+
+      const startResponse = apiServer.start();
+      expect(startResponse).not.toReject();
```

Comment:
@Leeyoungone This needs to be awaited for.

Reply:
got it! thank you king 👑